recovery_behavior_enabled: true
recovery_behavior: aggressive_reset
clearing_rotation_allowed: true
controller_frequency: 15.0

NavfnROS:
  allow_unknown: true # Specifies whether or not to allow navfn to create plans that traverse unknown space.
  default_tolerance: 0.5 # A tolerance on the goal point for the planner.
  use_dijkstra: false
  use_grid_path: true
  cost_factor: 5
  neutral_cost: 40
  lethal_cost: 253

DWAPlannerROS:
  ## Robot configuration parameters  
  acc_lim_x: 2.5
  acc_lim_y: 0
  acc_lim_th: 3.2

  max_vel_x: 1
  min_vel_x: 0.1
  max_vel_y: 0
  min_vel_y: 0

  max_trans_vel: 2
  min_trans_vel: 0.1
  max_rot_vel: 1.0
  min_rot_vel: 0.3

  ## Goal Tolerance Parameters
  yaw_goal_tolerance: 0.1
  xy_goal_tolerance: 0.25
  latch_xy_goal_tolerance: true

  ## Forward Simulation Parameters
  sim_time: 4.0
  # sim_granularity: 0.025
  vx_samples: 20
  vy_samples: 0
  vtheta_samples: 40
  # penalize_negative_x: false

  ## Trajectory scoring parameters
  path_distance_bias: 50.0 #32.0 # The weighting for how much the controller should stay close to the path it was given
  goal_distance_bias: 20.0 # The weighting for how much the controller should attempt to reach its local goal, also controls speed
  occdist_scale: 0.1 # The weighting for how much the controller should attempt to avoid obstacles
  # forward_point_distance: 0.325 # The distance from the center point of the robot to place an additional scoring point, in meters
  #stop_time_buffer: 0.3  # The amount of time that the robot must stop before a collision in order for a trajectory to be considered valid in seconds
  scaling_speed: .125 # The absolute value of the veolicty at which to start scaling the robot's footprint, in m/s
  # max_scaling_factor: 0.2 # The maximum factor to scale the robot's footprint by

  ## Oscillation Prevention Parameters
  # oscillation_reset_dist: 0.25 #How far the robot must travel in meters before oscillation flags are reset (double, default: 0.05)

